c/CMakeLists.txt

```diff
@@ -36,9 +36,21 @@ cmake_minimum_required(VERSION 3.19)
 # Project declaration needs to be before the configuration
 project(nvtx3 VERSION 3.1.1)
 
-option(NVTX3_INSTALL "Install NVTX3" ON)
+# Check if standalone or part of another project:
+set(NVTX_C_ROOT "${CMAKE_CURRENT_SOURCE_DIR}")
+string(REGEX REPLACE "/c$" "" NVTX_PROJ_ROOT "${NVTX_C_ROOT}")
+set(NVTX_TESTS_ROOT "${NVTX_PROJ_ROOT}/tests")
+set(NVTX_TOPLEVEL_PROJECT OFF)
+if ("${CMAKE_SOURCE_DIR}" STREQUAL "${NVTX_PROJ_ROOT}" OR
+    "${CMAKE_SOURCE_DIR}" STREQUAL "${NVTX_C_ROOT}" OR
+    "${CMAKE_SOURCE_DIR}" STREQUAL "${NVTX_TESTS_ROOT}")
+  set(NVTX_TOPLEVEL_PROJECT ON)
+endif()
```

Comment:
Should these new variable names be prefixed with `NVTX3_...` instead of `NVTX_...`? That would be needed for consistency with existing `NVTX3_...` variables. (This may apply other places in this PR too.)

Reply:
Also should we rename `TOPLEVEL_PROJECT` to `TOP_LEVEL_PROJECT`? (Use an underscore for each word break.)

Reply:
Good question. We have taken care to ensure the "3" is part of paths (`#include <nvtx3/nvToolsExt.h>`) and URLs (the `release-v3` branch), and I see use of `NVTX3` throughout our C++ bindings, but we use plain `NVTX_` in our C headers.

I would lean toward yes, we should use `NVTX3` here. It ultimately depends on what it would look like when NVTXv4 is published in the future and a project includes it its CMake but also includes a library via CMake that still uses NVTXv3.

CC @rdietrich-nvidia

Reply:
I support `TOP_LEVEL_PROJECT` with the underscore, which would match [`PROJECT_IS_TOP_LEVEL`](https://cmake.org/cmake/help/latest/variable/PROJECT_IS_TOP_LEVEL.html).

Reply:
Both changes sound good to me.